src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,174 @@
+import {
+  memo,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
 import { PeopleTable } from './PeopleTable';
+import { Sex } from '../types/SexEnum';
+import { SortType } from '../types/SortEnum';
+
+const toFindParents = (people: Person[], personName: string | null) => {
+  const personParent = people.find(({ name }) => name === personName);
+
+  return personParent;
+};
+
+export const PeoplePage = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasNoPeople, setHasNoPeople] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || Sex.ALL;
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const order = searchParams.get('order');
+  const sort = searchParams.get('sort');
+
+  const getPeopleFromServer = useCallback(async () => {
+    try {
+      const allPeople = await getPeople();
+      const peopleWithParents = allPeople.map((person) => {
+        return (
+          {
+            ...person,
+            mother: toFindParents(allPeople, person.motherName),
+            father: toFindParents(allPeople, person.fatherName),
+          }
+        );
+      });
+
+      if (allPeople.length) {
+        setPeople(peopleWithParents);
+      } else {
+        setHasNoPeople(true);
+      }
+
+      setIsLoading(false);
+    } catch {
+      setError(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    getPeopleFromServer();
+  }, []);
+
+  const filterBySex = useCallback((personSex: string) => {
+    return !sex
+      ? true
+      : personSex === sex;
+  }, [sex]);
+
+  const filterByQuery = useCallback((
+    name: string,
+    fatherName: string | null,
+    motherName: string | null,
+  ) => {
+    return name.toLowerCase().includes(query)
+    || fatherName?.toLowerCase().includes(query)
+    || motherName?.toLowerCase().includes(query);
+  }, [query]);
+
+  const filterByCenturies = useCallback((year: number) => {
+    return !centuries.length
+      ? true
+      : centuries.includes(String(Math.round(year / 100)));
```

Comment:
You can do it like this
```suggestion
    return !centuries.length || centuries.includes(String(Math.round(year / 100)));
```